### DISCUSSION TITLE
Incorrect bounds if a path has 0 length

### DISCUSSION THREAD

**Discussion by Joyphish** [vvvv/SVG#1202] [Q&A] [answered]
#### Description

SvgElement provides incorrect bounds if one path has 0 length

#### Example data

The bounds for the group in the following SVG is incorrect. The path with 0 length should be ignored when it comes to bounds.

```
<svg xmlns="http://www.w3.org/2000/svg" width="1056" height="816" viewBox="0 0 792 612">
<clipPath id="a"><path transform="matrix(1 0 0 -1 0 612)" d="M-396-306V918h1584V-306z"/></clipPath>
<g clip-path="url(#a)">
<path stroke-width="1.08" stroke-linecap="round" stroke-linejoin="round" fill="none" stroke="#000" d="M50 50 h100"/>
<path stroke-width="1.08" stroke-linecap="round" stroke-linejoin="round" fill="none" stroke="#000" d="M50 100 h0"/>
</g>
</svg>

```

#### Used Versions

SVG: 3.4.7
.Net Framework: 4.8.1
Operating system: Windows 11

**Comment by jlidbeck** [comment, marked as answer]
I believe this is expected behavior--[zero-length segments are not invalid](https://www.w3.org/TR/SVG2/paths.html#ZeroLengthSegments). The `stroke-linecap="round"` means that the linecaps will be rendered, this should affect the bounding rectangle.

> Note that when stroking a ‘[path](https://www.w3.org/TR/SVG2/paths.html#PathElement)’ element, any subpath consisting of a [moveto](https://www.w3.org/TR/SVG2/paths.html#PathDataMovetoCommands) but no following line drawing command will not be stroked. Any other type of zero-length subpath, such as 'M 10,10 L 10,10' or 'M 30,30 Z' will also not be stroked if the [stroke-linecap](https://www.w3.org/TR/SVG2/painting.html#StrokeLinecapProperty) property has a value of butt.

[https://www.w3.org/TR/SVG2/painting.html#SpecifyingStrokePaint](https://www.w3.org/TR/SVG2/painting.html#SpecifyingStrokePaint)
